### PR TITLE
[autoscaler] Better defaults and config options

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
Consider reviewing the Ray PR https://github.com/ray-project/ray/pull/26985 related to autoscaling defaults first.

## Why are these changes needed?

This PR wraps up autoscaler-related work for the Ray 2.0.0 release. Here's a summary of the changes

### Autoscaler container config specifcation.
Exposes Env and EnvFrom fields for the autoscaler container.

### Autoscaler default resources.
Sets the autoscaler default resource limits and requests **equal to each other** at 500m CPU and 512Mi memory.
Reasons for this
- It's easier to reason about pod scheduling with equal container limits and requests
- Having different values for container requests and resources automatically decreases the scheduling QoS class for the head pod, which we do not want. https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/

### Aggressive scaling.
Makes default scaling options more aggressive (https://github.com/ray-project/kuberay/issues/359)
Upscaling options are now "Conservative, Default, Aggressive" with Default and Aggressive being aliases of one another
(so upscaling is not rate-limited by default any more).

Completing this item will require merging the Ray PR https://github.com/ray-project/ray/pull/26985.
In fact, it might make sense to merge that one first.

### Revert volume mount change
This PR reverts the change to volume mounts from https://github.com/ray-project/kuberay/pull/391.
It turns out that this change can cause the Ray head container to crashloop with a file permissions error...
Having the Autoscaler container try to create the log file (https://github.com/ray-project/ray/pull/26748) if the Ray container hasn't already is good enough already.

We also remove a redundant volume mount from `raycluster-autoscaler.yaml` as it is already configured by the Ray operator.

### Note
Post-approval, please leave merging to me -- I might need to run some additional manual tests.

## Related issue number

Closes https://github.com/ray-project/kuberay/issues/358
Closes https://github.com/ray-project/kuberay/issues/359

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sriram-anyscale — COMMENTED** [review]
I don't feel qualified to approve this - but LGTM

File: ray-operator/apis/ray/v1alpha1/raycluster_types.go [hunk 1]
```diff
@@ -65,23 +65,28 @@ type ScaleStrategy struct {
 
 // AutoscalerOptions specifies optional configuration for the Ray autoscaler.
 type AutoscalerOptions struct {
-	// Resources specifies resource requests and limits for the autoscaler container.
-	// Default values: 256m CPU request, 512m CPU limit, 256Mi memory request, 512Mi memory limit.
+	// Resources specifies optional resource request and limit overrides for the autoscaler container.
+	// Default values: 500m CPU request and limit. 512Mi memory request and limit.
```
Comment (ericl) [inline ray-operator/apis/ray/v1alpha1/raycluster_types.go:69]: Is this per node? This is insanely low, typical Ray nodes should be 8-32GiB in size, or more than 10-40x this value. Ray is not designed to operate with tiny nodes.

Comment (DmitriGekhtman) [reply ray-operator/apis/ray/v1alpha1/raycluster_types.go:69]: These are defaults for the autoscaler container, not for a Ray node.

Comment (DmitriGekhtman) [reply ray-operator/apis/ray/v1alpha1/raycluster_types.go:69]: Important related discussion:
https://github.com/ray-project/kuberay/issues/417

File: ray-operator/apis/ray/v1alpha1/raycluster_types.go [hunk 2]
```diff
@@ -65,23 +65,28 @@ type ScaleStrategy struct {
...
 	Resources *v1.ResourceRequirements `json:"resources,omitempty"`
 	// Image optionally overrides the autoscaler's container image. This override is for provided for autoscaler testing and development.
 	Image *string `json:"image,omitempty"`
 	// ImagePullPolicy optionally overrides the autoscaler container's image pull policy. This override is for provided for autoscaler testing and development.
 	ImagePullPolicy *v1.PullPolicy `json:"imagePullPolicy,omitempty"`
+	// Optional list of environment variables to set in the autoscaler container.
+	Env []v1.EnvVar `json:"env,omitempty"`
+	// Optional list of sources to populate environment variables in the autoscaler container.
+	EnvFrom []v1.EnvFromSource `json:"envFrom,omitempty"`
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply ray-operator/apis/ray/v1alpha1/raycluster_types.go:78]: BTW, the overall change looks good to me and I just like to know when should we use these fields. like autoscaling config tuning etc

**Comment by Jeffwan** [comment]
please help run `./hack/update-codegen.sh` to make sure clientset is updated as well

File: ray-operator/apis/ray/v1alpha1/raycluster_types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (DmitriGekhtman) [reply ray-operator/apis/ray/v1alpha1/raycluster_types.go:78]: It's a good question:

There are some debug flags for certain things like 
- exposing detailed resource messages from the GCS
- modifying the autoscaler update interval

**Comment by DmitriGekhtman** [comment]
> ./hack/update-codegen.sh

We should probably add pre-push hooks for things like this.

**Comment by DmitriGekhtman** [comment]
Okidoki, thanks reviewers! Merging.

**Pull request merged**